src/jaegertracing/reporters/RemoteReporter.cpp

```diff
@@ -137,5 +141,14 @@ void RemoteReporter::flush()
     _lastFlush = Clock::now();
 }
 
+void RemoteReporter::flush()
+{
+    std::unique_lock<std::mutex> lock(_mutex);
+    _cv_flush.wait(lock, [this]() {
+        async_flush();
```

Comment:
This definitely looks wrong. The lambda here is just meant to express a boolean condition to avoid spurious wakeups. For example, instead of `while (running) { cv.wait(); }` you can do `cv.wait([&running] { return running; }`. This will potentially lead to many bugs in the program.